src/pal/test/snp_attestation_validation.cpp

```diff
@@ -82,6 +84,55 @@ TEST_CASE("Mismatched attestation and endorsements fail")
   }
 }
 
+TEST_CASE("Parsing of Tcb versions from strings")
+{
+  const auto milan_tcb_version_raw =
+    ccf::pal::snp::TcbVersionRaw::from_hex("d315000000000004");
+  CHECK_EQ(
+    nlohmann::json(milan_tcb_version_raw).dump(), "\"d315000000000004\"");
+
+  const auto milan_tcb_policy =
+    milan_tcb_version_raw.to_policy(ccf::pal::snp::ProductName::Milan);
+  CHECK_EQ(
+    nlohmann::json(milan_tcb_policy).dump(),
+    "{\"boot_loader\":4,\"hexstring\":\"d315000000000004\",\"microcode\":211,"
+    "\"snp\":21,\"tee\":0}");
```

Comment:
This is unnecessarily constrained, includes field order. We should construct an nlohmann::json object for equality, or check for presence-and-equality of each field that we care about.